tests/replication2.erl

```diff
@@ -11,592 +11,692 @@
-    ?assertEqual(0, repl_util:wait_for_reads(BSecond, 301, 400, TestBucket, 2)),
-
-    %% Testing fullsync with downed nodes
-    log_to_nodes(AllNodes, "Test fullsync with ~p and ~p down", [LeaderA, LeaderB]),
-    lager:info("Re-running fullsync with ~p and ~p down", [LeaderA, LeaderB]),
-
-    repl_util:start_and_wait_until_fullsync_complete(LeaderA2),
-
-    %%
-    %% Per-bucket repl settings tests
-    %%
-
-    log_to_nodes(AllNodes, "Test fullsync after restarting ~p", [LeaderA]),
-
-    lager:info("Restarting down node ~p", [LeaderA]),
-    rt:start(LeaderA),
-    rt:wait_until_pingable(LeaderA),
-    rt:wait_for_service(LeaderA, [riak_kv, riak_repl]),
-    repl_util:start_and_wait_until_fullsync_complete(LeaderA2),
-
-    log_to_nodes(AllNodes, "Starting Joe's Repl Test"),
-    lager:info("Starting Joe's Repl Test"),
-
-    %% @todo add stuff
-    %% At this point, realtime sync should still work, but, it doesn't because of a bug in 1.2.1
-    %% Check that repl leader is LeaderA
-    %% Check that LeaderA2 has ceeded socket back to LeaderA
-
-    lager:info("Leader: ~p", [rpc:call(ASecond, riak_core_cluster_mgr, get_leader, [])]),
-    lager:info("LeaderA: ~p", [LeaderA]),
-    lager:info("LeaderA2: ~p", [LeaderA2]),
-
-    ?assertEqual(ok, repl_util:wait_until_connection(LeaderA)),
-
-    log_to_nodes(AllNodes, "Simulate partition to force leader re-election"),
-
-    lager:info("Simulation partition to force leader re-election"),
-
-    OldCookie = rpc:call(LeaderA2, erlang, get_cookie, []),
-    NewCookie = list_to_atom(lists:reverse(atom_to_list(OldCookie))),
-    rpc:call(LeaderA2, erlang, set_cookie, [LeaderA2, NewCookie]),
-
-    [ rpc:call(LeaderA2, erlang, disconnect_node, [Node]) || Node <- ANodes -- [LeaderA2]],
-    [ rpc:call(Node, erlang, disconnect_node, [LeaderA2]) || Node <- ANodes -- [LeaderA2]],
-
-    repl_util:wait_until_new_leader(hd(ANodes -- [LeaderA2]), LeaderA2),
-    InterimLeader = rpc:call(LeaderA, riak_core_cluster_mgr, get_leader, []),
-    lager:info("Interim leader: ~p", [InterimLeader]),
-
-    %rpc:call(LeaderA2, erlang, apply, [fun() -> [net_adm:ping(N) || N <- ANodes] end, []]),
-    rpc:call(LeaderA2, erlang, set_cookie, [LeaderA2, OldCookie]),
+          ok
+      end,
+
+      lager:info("Check keys written before repl was connected are present"),
+      ?assertEqual(0, repl_util:wait_for_reads(BFirst, 1, 200, TestBucket, 2));
+    _ ->
+      ok
+  end.
+
+
+%% @doc Disconnected Clusters Full Sync Test
+%%      Test Cycle:
+%%        Disconnect Clusters "A" and "B".
+%%        Write 2000 keys to Cluster "A".
+%%        Reconnect Clusters "A" and "B" and enable real time and full sync.
+%%        Read 2000 keys from Cluster "B".
+disconnected_cluster_fsync_test( [AFirst|_] = ANodes, [BFirst|_] = BNodes ) ->
+  
+  TestHash =  list_to_binary([io_lib:format("~2.16.0b", [X]) || <<X>> <= erlang:md5(term_to_binary(os:timestamp()))]),
```

Comment:
Erlang spacing is usually 4 spaces.  Can we adjust this to reduce the amount of whitespace changes in this diff?